sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -439,4 +439,53 @@ namespace Azure { namespace Core { namespace Test {
     CheckBodyFromBuffer(*response, expectedResponseBodySize);
   }
 
+  TEST_F(TransportAdapter, cancelTransferUpload)
+  {
+    Azure::Core::Http::Url host("http://httpbin.org/put");
+    Azure::Core::Context cancelThis;
+
+    auto threadRoutine = [host, cancelThis]() {
+      // Start a big upload and expect it to throw cancelation
+      std::vector<uint8_t> bigBuffer(1024 * 1024 * 200, 'x'); // upload 200 Mb
+      auto stream = Azure::Core::Http::MemoryBodyStream(bigBuffer);
+      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Put, host, &stream);
+
+      // Request will be canceled from main thread throwing the excpetion
+      EXPECT_THROW(pipeline.Send(cancelThis, request), Azure::Core::RequestCanceledException);
+    };
+
+    // Start request
+    std::thread t1(threadRoutine);
+
+    // Wait 100 ms so we know upload has started
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+    cancelThis.Cancel();
+    t1.join();
+  }
+
+  TEST_F(TransportAdapter, cancelTransferDownload)
+  {
+    // ~50Mb text
+    Azure::Core::Http::Url host("https://raw.githubusercontent.com/vhvb1989/"
```

Comment:
@weshaggard - we want to download some big payload from a url/server to test request cancellation, mid-way through. Do you know of a good way to do this as part of basic unit tests (without needed to setup a storage client/upload or the live testing approach)?

Can we upload/check-in this file to an azure sdk bot GitHub account?

Reply:
If we need this we can put it on a static blob container that you can try to download. However we should be somewhat careful about trying to download large files all the time.

Reply:
I should say we should definitely avoid committing a large file into the repo, we don't what that in our git history as everyone will have to pay that cost of downloading it forever.

Reply:
yes, I don't want to add the file to the repo.
Initially I used another public repo to host the file. github max size for files is 50MB.
So then I better moved the file to Azure Storage account. With anonymous access to blobs only.  There we can have bigger files.
We don't want to download the file every time. We want to cancel the download during the process and validate this can be done.